File: num287.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        length = len(nums)
        dp = [0 for _ in range(length+1)]
        for item in nums:
            dp[item] += 1
            if dp[item] > 1:
                return item

if __name__ == "__main__":
    solution = Solution()
    nums = [1,3,4,2,2]
    print(solution.findDuplicate(nums))

File: daily_problem/num287.java
package daily_problem;

import java.util.Arrays;
import java.util.List;

public class num287 {
    public int findDuplicate(int[] nums) {
        int item = 0;
        for(int num:nums){
            item ^= num;
        }
        return item;
    }

    public static void main(String[] args) {
        String s = "a, b, c, d";
        List<String> list = Arrays.asList(s.split(","));
        System.out.println(StringUtils.join(list.toArray(), ","));
    }
}